CONTRIBUTING.md

```diff
@@ -21,14 +21,90 @@ Unsure where to begin contributing? You can start by looking through these `begi
-4. Write appropriate comments in the pull request, explaining the changes made. This helps maintainers, and the community understand the changes made by you and help review the PR. **Reference the issue you are fixing**.
-5. For changes to the front-end include screenshots or screencast showing the working changes. You can use browser extensions like `Screencastify` to create a screencast.
-6. Make sure you submit some kind of proof of your fix that makes the reviewing process easier for the maintainers.
+### Best Practices to send Pull Requests:
+
+  - Fork the [project](https://github.com/bench-routes/bench-routes) on GitHub
+  - Clone the project locally into your system.
+```
+git clone https://github.com/your-username/bench-routes.git
+```
+  - Make sure you are in the `master` branch.
+```
+git checkout master
+```
+  - Create a new branch with a meaningful name for each feature/pull request. 
+```
+git checkout -b branch-name
+```
+  - Make all the required changes.
+
+  - Ensure that the code follows the linting standards. `eslint`, `tslint` and `prettier` are used for the TypeScript files and `golint` is used for the Golang files. Use the command `make fix` to format the code.
+  - Add the files you changed.
+```
+git add file-name
+```
+  - Write appropriate comments in the pull request, explaining the changes made. This helps maintainers, and the community understand the changes made by you and help review the PR. **Reference the issue you are fixing**.
+```
+git commit -m "message"
+```
+  - If you forgot to add some changes, you can edit your previous commit message.
+```
+git commit --amend
+```
+  - Squash multiple commits to a single commit. (example: squash last two commits done on this branch into one)
+```
+git rebase --interactive HEAD~2 
+```
+  - Push this branch to your remote repository on GitHub.
+```
+git push origin branch-name
+```
+  - If any of the squashed commits have already been pushed to your remote repository, you need to do a force push.
+```
+git push origin remote-branch-name --force
+```
+
+  - For changes to the front-end include screenshots or screencast showing the working changes. You can use browser extensions like `Screencastify` to create a screencast.
+
+  - Make sure you submit some kind of proof of your fix that makes the reviewing process easier for the maintainers.
+
+  - During review, if you are requested to make changes, rebase your branch and squash the multiple commits into one. Once you push these changes the pull request will edit automatically.
+
+## Configuring remotes
+When a repository is cloned, it has a default remote called `origin` that points to your fork on GitHub, not the original repository it was forked from. To keep track of the original repository, you should add another remote called `upstream`.
```

Comment:
I think, if someone clones from the original repo, then origin will point to the original repo, and not fork. I think this line is misleading. It should be like origin is what you have cloned. Make sure that your clone is the fork and not benchroutes origin.

Reply:
The contribution.md is mostly for the external contributors and not the maintainer of the project and the contributors don't have access to create a branch or push any code to the original repo. So, I have written them from their perspective, who ever wants to make a contribution, they have to fork the repo for sure.